### ISSUE TITLE
Schema/Owner Table Config Support

### ISSUE THREAD

**Issue by davious** [rebus-org/Rebus.PostgreSql#11]
Is there a way to configure the owner/schema of the rebus tables?
If not, would you add that configuration option?

btw ❤️ and 👏 rebus and this plugin

**Comment by mookid8000** [comment]
Hi @davious , I'm so sorry about the late reply!!

I am not that familiar with PostgreSQL... what is "the owner/schema of the schema/owner"?

And yes, I am definitely open to adding whichever configuration is needed to make the Postgres integration align better with common usage 🙂

**Comment by davious** [comment]
> Hi @davious , I'm so sorry about the late reply!!
> 
> I am not that familiar with PostgreSQL... what is "the owner/schema of the schema/owner"?
> 
> And yes, I am definitely open to adding whichever configuration is needed to make the Postgres integration align better with common usage 🙂

Hi Mogens,

It's just the schema. I think in postgres it's `public` by default.

`SELECT * FROM messages` would work -- and `SELECT * FROM public.messages` would also work.

In SQL Server, the default is `dbo`.

And, what I what to do is keep the rebus tables in a separate `rebus` schema within the same database -- so it is nicely separated from my application's tables.

The concept is simple, my typo made it confusing. Sorry about that - I've updated my question -- "the owner/schema of the schema/owner" -> "the owner/schema".

**Comment by mookid8000** [comment]
Ah, it makes sense then 😁 for Rebus' SQL Server transport I am currently working on making it possible to configure a schema for the transport, which will work as a namespace of sorts for all of the queue names.

The current version of the SQL Server transport accepts fully qualified queue names on the form `[dbo].[my-queue]`, `[rebus].[another-queue]`, etc., but it's actually kind of cumbersome to support that.... also it made the code using those names very hard to port to other queueing systems, so on [this branch](https://github.com/rebus-org/Rebus.SqlServer/tree/feature/new-sql-transport) I started working on cleaning up the SQL transport.

With the new transport, you will configure the schema like this:
```csharp
Configure.With(...)
	.Transport(t => t.UseSqlServer(connectionString, "my-queue")
		.WithSchema("rebus"))
	.Start();
```

Do you think something like that would work for the Postgres transport?

**Comment by davious** [comment]
> Do you think something like that would work for the Postgres transport?

Yes. Precisely. 👌

**Comment by zabulus** [comment]
my 50 cents: SqlServer works if you specify <schemaname>.my-queue without any additional effort.

**Comment by patrick11994** [comment]
This issue is resolved by PR #33 🚀
